### ISSUE TITLE
Empty tuple initialization cannot be extracted as function

### ISSUE THREAD

**Issue by omertuc** [rust-lang/rust-analyzer#15820]
**rust-analyzer version**: `rust-analyzer 1.73.0 (cc66ad4 2023-10-03)`

**rustc version**: `rustc 1.73.0 (cc66ad468 2023-10-03)`

**relevant settings**: n/a

# Input

```rust
struct Foo;
fn foo() -> Foo {
    $0Foo$0
}
```

# Expected

Be given the option to extract the Foo expression into a function

# Actual

No option given

# Explanation

If this was a regular struct instead then the option is given:

```rust
struct Foo{}
fn foo() -> Foo {
    $0Foo{}$0
}
```

turns to:

```rust
struct Foo{}
fn foo() -> Foo {
    fun_name()
}

fn fun_name() -> Foo {
    Foo{}
}
```

**Comment by omertuc** [comment]
@ChayimFriedman2 I believe this issue should be reopened, as it calls for having the ability to extract as a function, but it seems it's only possible to extract as a variable following the PR

<img width="516" height="436" alt="Image" src="https://github.com/user-attachments/assets/3c840f4e-815e-40a2-a375-84542f8739eb" />

**Comment by ChayimFriedman2** [comment]
Right, although that might just be a typo. I don't know if there is benefit in extracting this to function, but I won't object a PR if it will be small.
